src/coreclr/System.Private.CoreLib/src/System/InvokeUtils.cs

```diff
@@ -0,0 +1,303 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    internal static class InvokeUtils
```

Comment:
Can this live in https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Reflection/InvokeUtils.cs ?

Reply:
The InvokeUtils under shared corelib isn't used by other runtimes currently. Nothing prevents it to work since `CorElementType` is available.

Exposing it to naot may cause name collision around type and methods. Currently it's not an issue because of namespace different (System for naot and System.Refection for shared).

Since it's not ready to be shared between runtimes, I'd prefer to put it under coreclr for now.

Reply:
> Exposing it to naot may cause name collision

It won't. The shared InvokeUtils.cs is excluded when compiling CoreLib for native AOT currently.

> Since it's not ready to be shared between runtimes,

The shared InvokeUtils.cs is prepared for being shared eventually, but it is currently only used by regular CoreCLR.

Reply:
What would be its namespace? I think System makes more sense since the only Reflection type used is Pointer now.

Reply:
System.Reflection just like the existing shared InvokeUtils. Both existing shared System.Reflection.InvokeUtils and the code you are in this PR implement reflection-specific primitive type widening rules.